### ISSUE TITLE
SQLite 3 Exception

### ISSUE THREAD

**Issue by GlassAndOneHalf** [layday/instawow#45]
Hi there! I recently updated my instawow from `v1.12.1` to `v1.13.3` and when I attempted to update my WoW classic addons I got this error:
```
PS C:\Users\Jacob> instawow -p classic update
Traceback (most recent call last):
  File "c:\users\jacob\scoop\apps\python\3.8.5\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "c:\users\jacob\scoop\apps\python\3.8.5\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\Jacob\scoop\apps\python\current\Scripts\instawow.exe\__main__.py", line 7, in <module>
  File "C:\Users\Jacob\AppData\Roaming\Python\Python38\site-packages\click\core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "C:\Users\Jacob\AppData\Roaming\Python\Python38\site-packages\click\core.py", line 782, in main
    rv = self.invoke(ctx)
  File "C:\Users\Jacob\AppData\Roaming\Python\Python38\site-packages\click\core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "C:\Users\Jacob\AppData\Roaming\Python\Python38\site-packages\click\core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "C:\Users\Jacob\AppData\Roaming\Python\Python38\site-packages\click\core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "C:\Users\Jacob\AppData\Roaming\Python\Python38\site-packages\click\decorators.py", line 33, in new_func
    return f(get_current_context().obj, *args, **kwargs)
  File "c:\users\jacob\scoop\apps\python\3.8.5\lib\site-packages\instawow\cli.py", line 306, in update
    Report(results.items(), filter_results).generate_and_exit()
  File "c:\users\jacob\scoop\apps\python\3.8.5\lib\site-packages\instawow\cli.py", line 91, in generate_and_exit
    self.generate()
  File "c:\users\jacob\scoop\apps\python\3.8.5\lib\site-packages\instawow\cli.py", line 86, in generate
    report = str(self)
  File "c:\users\jacob\scoop\apps\python\3.8.5\lib\site-packages\instawow\cli.py", line 72, in __str__
    return '\n'.join(
  File "c:\users\jacob\scoop\apps\python\3.8.5\lib\site-packages\instawow\cli.py", line 74, in <genexpr>
    + fill(r.message, initial_indent=' ' * 2, subsequent_indent=' ' * 4)
  File "c:\users\jacob\scoop\apps\python\3.8.5\lib\site-packages\instawow\exceptions.py", line 16, in message
    return self.message_template.format(self=self)
  File "c:\users\jacob\scoop\apps\python\3.8.5\lib\site-packages\sqlalchemy\orm\attributes.py", line 287, in __get__
    return self.impl.get(instance_state(instance), dict_)
  File "c:\users\jacob\scoop\apps\python\3.8.5\lib\site-packages\sqlalchemy\orm\attributes.py", line 718, in get
    value = state._load_expired(state, passive)
  File "c:\users\jacob\scoop\apps\python\3.8.5\lib\site-packages\sqlalchemy\orm\state.py", line 652, in _load_expired
    self.manager.deferred_scalar_loader(self, toload)
  File "c:\users\jacob\scoop\apps\python\3.8.5\lib\site-packages\sqlalchemy\orm\loading.py", line 1008, in load_scalar_attributes
    result = load_on_ident(
  File "c:\users\jacob\scoop\apps\python\3.8.5\lib\site-packages\sqlalchemy\orm\loading.py", line 201, in load_on_ident
    return load_on_pk_identity(
  File "c:\users\jacob\scoop\apps\python\3.8.5\lib\site-packages\sqlalchemy\orm\loading.py", line 287, in load_on_pk_identity
    return q.one()
  File "c:\users\jacob\scoop\apps\python\3.8.5\lib\site-packages\sqlalchemy\orm\query.py", line 3458, in one
    ret = self.one_or_none()
  File "c:\users\jacob\scoop\apps\python\3.8.5\lib\site-packages\sqlalchemy\orm\query.py", line 3427, in one_or_none
    ret = list(self)
  File "c:\users\jacob\scoop\apps\python\3.8.5\lib\site-packages\sqlalchemy\orm\query.py", line 3503, in __iter__
    return self._execute_and_instances(context)
  File "c:\users\jacob\scoop\apps\python\3.8.5\lib\site-packages\sqlalchemy\orm\query.py", line 3524, in _execute_and_instances
    conn = self._get_bind_args(
  File "c:\users\jacob\scoop\apps\python\3.8.5\lib\site-packages\sqlalchemy\orm\query.py", line 3539, in _get_bind_args
    return fn(
  File "c:\users\jacob\scoop\apps\python\3.8.5\lib\site-packages\sqlalchemy\orm\query.py", line 3518, in _connection_from_session
    conn = self.session.connection(**kw)
  File "c:\users\jacob\scoop\apps\python\3.8.5\lib\site-packages\sqlalchemy\orm\session.py", line 1138, in connection
    return self._connection_for_bind(
  File "c:\users\jacob\scoop\apps\python\3.8.5\lib\site-packages\sqlalchemy\orm\session.py", line 1146, in _connection_for_bind
    return self.transaction._connection_for_bind(
  File "c:\users\jacob\scoop\apps\python\3.8.5\lib\site-packages\sqlalchemy\orm\session.py", line 409, in _connection_for_bind
    self._assert_active()
  File "c:\users\jacob\scoop\apps\python\3.8.5\lib\site-packages\sqlalchemy\orm\session.py", line 289, in _assert_active
    raise sa_exc.InvalidRequestError(
sqlalchemy.exc.InvalidRequestError: This Session's transaction has been rolled back due to a previous exception during flush. To begin a new transaction with this Session, first issue Session.rollback(). Original exception was: (raised as a result of Query-invoked autoflush; consider using a session.no_autoflush block if this flush is occurring prematurely)
(sqlite3.IntegrityError) UNIQUE constraint failed: pkg_dep.id, pkg_dep.pkg_source, pkg_dep.pkg_id
[SQL: INSERT INTO pkg_dep (id, pkg_source, pkg_id) VALUES (?, ?, ?)]
[parameters: ('327139', 'curse', '335852')]
(Background on this error at: http://sqlalche.me/e/13/gkpj) (Background on this error at: http://sqlalche.me/e/13/7s2a)
```

Now when I try to update again, I get this error:
```
PS C:\Users\Jacob> instawow --debug -p classic update
! curse:guidelime-busteas-1-60-leveling
  internal error
! curse:characterstatsclassic
  internal error
! curse:atlaslootclassic
  internal error
! curse:tradeskill-master
  internal error
! curse:deadly-boss-mods
  internal error
```

These are my installed addons:
```
PS C:\Users\Jacob> instawow -p classic list
curse:atlaslootclassic
curse:best-in-slot-classic
curse:characterstatsclassic
curse:deadly-boss-mods
curse:guidelime
curse:guidelime-busteas-1-60-leveling
curse:guidelime_sage
curse:omni-cc
curse:pally-power
curse:questie
curse:total-rp-3-classic
curse:tradeskill-master
tukui:2-elvui
wowi:25006-classicauradurations
```

This is using a non-admin Powershell on Windows 10 19041. Interestingly, updating my retail addons occurred without issue.

**Comment by layday** [comment]
Thanks for the report.  This occurs when attempting to update `curse:guidelime-busteas-1-60-leveling` which depends on `curse:guidelime`.  It would appear that because the dependency table uses a `UNIQUE` constraint, our SQL library issues an `INSERT` instead of an `UPDATE` statement when updating `curse:guidelime-busteas-1-60-leveling` to the latest version, and because the row already exists, well...

**Comment by layday** [comment]
Though I am curious why the rest of them are erroring.  Could you paste the contents of your `error.log`?  Run `instawow -p classic configure --active`; your error log will be under the `config_dir` then 'profiles/classic/logs/error.log'.

**Comment by GlassAndOneHalf** [comment]
Of course! It's quite long so I've added it as an attachment here ([error.log](https://github.com/layday/instawow/files/5188353/error.log)). Thanks for taking a look into this 💯

**Comment by layday** [comment]
Cheers.  It looks like `curse:guidelime-busteas-1-60-leveling` puts the database in an invalid state when it fails so the rest of the updates cannot proceed.

**Comment by layday** [comment]
Fixed in v1.14.1.  Thanks again.
